src/histogram-date-range.ts

```diff
@@ -163,32 +255,113 @@ export class HistogramDateRange extends LitElement {
-    window.addEventListener('pointercancel', this.drop);
+    this._isDragging = true;
+    this.addListeners();
+    this.cancelPendingUpdateEvent();
   };
 
   private drop = (): void => {
-    this.isDragging = false;
-
-    window.removeEventListener('pointermove', this.move);
-    window.removeEventListener('pointerup', this.drop);
-    window.removeEventListener('pointercancel', this.drop);
+    if (this._isDragging) {
+      this.removeListeners();
+      this.beginEmitUpdateProcess();
+    }
+    this._isDragging = false;
   };
 
+  /**
+   * Adjust the date range based on slider movement
+   *
+   * @param e PointerEvent from the slider being moved
+   */
   private move = (e: PointerEvent): void => {
     const newX = e.offsetX - this._dragOffset;
     const slider = this._currentSlider as SVGRectElement;
-    return (slider.id as SliderId) === 'slider-min'
-      ? this.setMinSlider(newX)
-      : this.setMaxSlider(newX);
+    const date = this.translatePositionToDate(newX);
+    if ((slider.id as SliderId) === 'slider-min') {
+      this.minSelectedDate = date;
+    } else {
+      this.maxSelectedDate = date;
+    }
   };
 
-  // find position of pointer in relation to the current slider
+  /**
+   * Constrain a proposed value for the minimum (left) slider
+   *
+   * If the value is less than the leftmost valid position, then set it to the
+   * left edge of the widget (ie the slider width). If the value is greater than
+   * the rightmost valid position (the position of the max slider), then set it
+   * to the position of the max slider
+   */
+  private validMinSliderX(newX: number): number {
+    const validX = Math.max(newX, this.sliderWidth);
+    return Math.min(validX, this.maxSliderX);
+  }
+
+  /**
+   * Constrain a proposed value for the maximum (right) slider
+   *
+   * If the value is greater than the rightmost valid position, then set it to
+   * the right edge of the widget (ie widget width - slider width). If the value
+   * is less than the leftmost valid position (the position of the min slider),
+   * then set it to the position of the min slider
+   */
+  private validMaxSliderX(newX: number): number {
+    const validX = Math.max(newX, this.minSliderX);
+    return Math.min(validX, this.width - this.sliderWidth);
+  }
+
+  private addListeners(): void {
+    window.addEventListener('pointermove', this.move);
+    window.addEventListener('pointerup', this.drop);
+    window.addEventListener('pointercancel', this.drop);
+  }
+
+  private removeListeners(): void {
+    window.removeEventListener('pointermove', this.move);
+    window.removeEventListener('pointerup', this.drop);
+    window.removeEventListener('pointercancel', this.drop);
+  }
+
+  /**
+   * start a timer to emit an update event. this timer can be canceled (and the
+   * event not emitted) if user drags a slider or focuses a date input within
+   * the update delay
+   */
+  private beginEmitUpdateProcess(): void {
```

Comment:
Side thought, but we should add throttler/debouncer cancellation support in our [throttler/debouncer module](https://github.com/internetarchive/iaux-throttler-debouncer). You could've used it instead of making a debouncer.

Reply:
Yeah, the cancel ability was crucial. I also wanted to be able to specify a custom value for the debounce delay

Reply:
I guess maybe I should've said this as a suggestion 😅 What do you think about updating the debouncer package to support cancellation? It supports it internally, it's just not exposed. It'd be great if we had a generally useful debouncer/throttler package so we don't recreate debouncers in every project we work on.

Reply:
Agreed, having a good generalized debouncer would be nice. I don't have the bandwidth to do it now though but would be happy to later on :)